### PR TITLE
EZP-30712: Fixed reverseTransform for midnight value

### PR DESCRIPTION
**Pull request by SerheyDolgushev**
**JIRA**: https://jira.ez.no/browse/EZP-30712

Can't set a time to be midnight without this fix

### CODE REVIEW

**Comment by micszo** [comment]
Hi @SerheyDolgushev, thanks for opening the PR! Could you change base and rebase to 2.5?

File: lib/FieldType/DataTransformer/TimeValueTransformer.php [hunk 1]
```diff
@@ -50,7 +50,7 @@ public function transform($value)
      */
     public function reverseTransform($value)
     {
-        if (empty($value)) {
+        if (is_null($value) || $value === '') {
```
Comment (alongosz) [inline lib/FieldType/DataTransformer/TimeValueTransformer.php]: CS:
```suggestion
        if (null === $value || $value === '') {
```

Comment (SerheyDolgushev) [reply lib/FieldType/DataTransformer/TimeValueTransformer.php]: Done

**Comment by SerheyDolgushev** [comment]
Requested CS changes and rebase against 2.5 is done.

`TimeValueTransformer` uses `eZ\Publish\Core\FieldType\Time\Value`, which is not the part of this package. So there is no easy way to test `TimeValueTransformer`. @alongosz can you please suggest a solution here?

**Comment by alongosz** [comment]
This is still unresolved:

> @SerheyDolgushev is it possible to cover this with some minimal unit tests? I'd like to know what empty-ish value occurs for midnight w/o debugging it myself (good input data for test case) ;)
> 
> Seems we're missing tests for `TimeValueTransformer` so please add minimal coverage of just `testTransform` with a few inputs. Please see how it's done in `\EzSystems\RepositoryForms\Tests\FieldType\DataTransformer\DateIntervalToArrayTransformerTest`.

I'm sorry, but I don't have time to reproduce this myself and decide if the fix is correct based on behavior. This is why tests are helpful.

**Comment by SerheyDolgushev** [comment]
@alongosz please review https://github.com/ezsystems/repository-forms/pull/301/commits/0e9d3b3f7a837272646596bb8745df8c3c0148f2.

File: tests/RepositoryForms/FieldType/DataTransformer/TimeValueTransformerTest.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        $transformer = new TimeValueTransformer();
+
+        self::assertNull($transformer->transform($value));
+    }
+
+    public function testTransformInvalidValue()
+    {
+        $transformer = new TimeValueTransformer();
+
+        self::expectException(TransformationFailedException::class);
```
Comment (alongosz) [inline tests/RepositoryForms/FieldType/DataTransformer/TimeValueTransformerTest.php]: AFAICS `expectException` is not static, so should be refered by `$this->`

File: tests/RepositoryForms/FieldType/DataTransformer/TimeValueTransformerTest.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+        self::assertSame($time, $transformer->transform($value));
+    }
+
+    public function testTransformZero()
+    {
+        $value = new Value(0);
+        $transformer = new TimeValueTransformer();
+
+        self::assertSame(0, $transformer->transform($value));
```
Comment (alongosz) [inline tests/RepositoryForms/FieldType/DataTransformer/TimeValueTransformerTest.php]: Ok, now I get the fix, thanks!

File: tests/RepositoryForms/FieldType/DataTransformer/TimeValueTransformerTest.php (hunk 2 above)
Comment (SerheyDolgushev) [reply tests/RepositoryForms/FieldType/DataTransformer/TimeValueTransformerTest.php]: Done: https://github.com/ezsystems/repository-forms/pull/301/commits/da92b939fe773a623e39b5d9f3775082aedebab8

**Comment by lserwatka** [comment]
@alongosz could you merge it up?

**Comment by alongosz** [comment]
@lserwatka done.

Thank you @SerheyDolgushev :tada:
